### PR TITLE
schema: Document that names should be fully-qualified.

### PR DESCRIPTION
**Pull request by varungandhi-src**
I know it's kinda' "obvious" if you think about it for a bit, but I made
the mistake of following the words very literally and ended up having
a silly bug in scip-ruby with the first implementation. https://github.com/sourcegraph/scip-ruby/issues/14

### Test plan

n/a

### CODE REVIEW

File: scip.proto [hunk 1]
```diff
@@ -119,6 +119,12 @@ message Document {
 //   <escaped-identifier>   ::= '`' { <escaped-character> } '`'
 //   <escaped-characters>   ::= any UTF-8 character, escape backticks with double backtick.
 // ```
+//
+// For "top-level" identifiers, such as types, fields/terms, methods, the
+// `<name>` should be a fully-qualified within the project. Generally, this
```
Comment (olafurpg) [inline scip.proto:124]: This sounds inaccurate, the `Symbol.descriptors` field (which maps to `{ <descriptor> }`) should be fully qualified while `Descriptor.name` should be a single identifier like the name of a method or field 🤔

Comment (varungandhi-src) [reply scip.proto:124]: Ah; it's the same confusion from before around the semantics of `{ }`. I'll replace that with a regex `+` and document that.

Comment (varungandhi-src) [reply scip.proto:124]: It seems like the namespace can/should contain the filepath though? Or do we not rely on that? That's not necessarily obvious from the text.

Comment (olafurpg) [reply scip.proto:124]: The descriptor doesn't need to include a file path, for JVM languages we only use packages (example `java/io/File#`). The only hard requirement is that the descriptor must be stable between compilations and match at the definition site and at reference site (which can be a separate repo from the definition). It's desirable, but optional, that the descriptor has human readable names.

Comment (varungandhi-src) [reply scip.proto:124]: So the complete list of descriptors has to be "fully qualified" when viewed as a whole, right?

In Java's case, IIUC, the path essentially ends up being encoded in the package structure anyways, so a separate path is not necessary.

Comment (olafurpg) [reply scip.proto:124]: > So the complete list of descriptors has to be "fully qualified" when viewed as a whole, right?

Correct.

Comment (varungandhi-src) [reply scip.proto]: Updated the wording; can you change a second look?

**Review by olafurpg — APPROVED** [review]
LGTM 👍 I like the new clarification, and `(PATTERN)+` syntax. The `{PATTERN}` syntax was borrowed from the Scala parser

**Pull request merged**